### PR TITLE
fix(ui): Fix SSO-required infinite redirect

### PR DESCRIPTION
**Pull request by billyvg**

### CODE REVIEW

File: src/sentry/api/bases/organization.py [hunk 1]
```diff
@@ -74,7 +74,10 @@ def has_object_permission(self, request, view, organization):
                             'user_id': request.user.id,
                         }
                     )
-                    raise NotAuthenticated(detail='Must login via SSO')
+                    raise NotAuthenticated(detail={
```
Comment (billyvg) [inline src/sentry/api/bases/organization.py]: I don't think we use `detail` anywhere

**Review by dcramer — APPROVED** [review]
this is fine but detail has always been a string so I'd probably just put the keys top level. This would line it up with how others work.

```
data = {
                "error": "Account verification required.",
                "sudoRequired": True,
                "username": request.user.username,
            }
            return Response(data, status=401)
```

**Comment by dcramer** [comment]
Also I would return the absolute URL to authenticate as part of the payload, given it is a server-rendered view.

**Comment by billyvg** [comment]
Looks like `NotAuthenticated` extends `APIException` which only takes `detail`


```
    def __init__(self, detail=None):
        self.detail = detail or self.default_detail
```

**Comment by dcramer** [comment]
@billyvg yeah id just say change that to be more useful/future proof

we wrote this up at one point but haven't done shit with it https://docs.sentry.io/internal/api/

File: src/sentry/api/exceptions.py [hunk 2]
```diff
@@ -1,14 +1,55 @@
...
+
+    def __init__(self, organization):
+        detail = {
+            'code': 'sso-required',
+            'message': 'Must login via SSO',
+            'extra': {
+                'loginUrl': reverse('sentry-auth-organization', args=[organization.slug]),
+            }
+        }
+        super(SsoRequired, self).__init__(detail=detail)
```
Comment (billyvg) [inline src/sentry/api/exceptions.py]: How does this look @dcramer ?

Comment (dcramer) [reply src/sentry/api/exceptions.py]: guccigang

**Review by dcramer — APPROVED** [review]

**Pull request merged**